ui/desktop/src/utils/shims.ts

```diff
@@ -0,0 +1,16 @@
+export const SUPPORTED_EXTENSION_RUNNERS = ['jbang', 'npx', 'uvx'];
+export const SHIM_COMMANDS = ['goosed', ...SUPPORTED_EXTENSION_RUNNERS];
+
+/**
+ * Replaces the command with the corresponding binary path for the shim if one is supported
+ * @param cmd The command to potentially replace with a shim
+ * @returns The path to the binary if it's a known shim, otherwise returns the original command
+ */
+export async function replaceWithShims(cmd: string): Promise<string> {
+  if (SHIM_COMMANDS.indexOf(cmd) === -1) {
+    return cmd;
+  }
+  const binaryPath = window.electron.getBinaryPath(cmd);
+  console.log('--------> Replacing command with shim ------>', cmd, binaryPath);
+  return binaryPath;
```

Comment:
In addition to centralizing the implementation, I've also simplified it. The previous implementation would build a dictionary that included calls to `window.electron.getBinary` for _all_ shims, regardless of which shim command was requested. This implementation only calls that function for the specific shim command requested.